src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -305,6 +305,10 @@ protected function normalizeGroups($groups)
     private function validateObject($object, string $propertyPath, array $groups, int $traversalStrategy, ExecutionContextInterface $context)
     {
         try {
+            if (!\is_object($object)) {
+                throw new NoSuchMetadataException(sprintf('Cannot create metadata for non-objects. Got: "%s".', \gettype($object)));
```

Comment:
TBF, the validation layer should not throw here, but reject the payload with a validation error, but I suppose "one thing at a time"